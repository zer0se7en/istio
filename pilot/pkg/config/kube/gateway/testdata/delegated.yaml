apiVersion: networking.x-k8s.io/v1alpha1
kind: GatewayClass
metadata:
  name: istio
spec:
  controller: istio.io/gateway-controller
---
apiVersion: networking.x-k8s.io/v1alpha1
kind: Gateway
metadata:
  name: gateway
  namespace: istio-system
spec:
  gatewayClassName: istio
  listeners:
  - hostname: "apple.example"
    port: 80
    protocol: HTTP
    routes:
      namespaces:
        from: Selector
        selector:
          matchLabels:
            kubernetes.io/metadata.name: apple
      kind: HTTPRoute
  - hostname: "banana.example"
    port: 80
    protocol: HTTP
    routes:
      namespaces:
        from: Selector
        selector:
          matchLabels:
            kubernetes.io/metadata.name: banana
      kind: HTTPRoute
---
apiVersion: networking.x-k8s.io/v1alpha1
kind: HTTPRoute
metadata:
  name: http
  namespace: apple
spec:
  gateways:
    allow: All
  rules:
  - forwardTo:
    - serviceName: httpbin-apple
      port: 80
---
apiVersion: networking.x-k8s.io/v1alpha1
kind: HTTPRoute
metadata:
  name: http
  namespace: banana
spec:
  gateways:
    allow: All
  rules:
  - forwardTo:
    - serviceName: httpbin-banana
      port: 80
---