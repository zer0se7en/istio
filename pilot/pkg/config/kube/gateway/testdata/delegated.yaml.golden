apiVersion: networking.istio.io/v1alpha3
kind: Gateway
metadata:
  annotations:
    internal.istio.io/gateway-service: istio-ingressgateway.istio-system.svc.domain.suffix
  creationTimestamp: null
  name: gateway-istio-autogenerated-k8s-gateway
  namespace: istio-system
spec:
  servers:
  - hosts:
    - apple.example
    port:
      name: 0-gateway-gateway-istio-system
      number: 80
      protocol: HTTP
  - hosts:
    - banana.example
    port:
      name: 1-gateway-gateway-istio-system
      number: 80
      protocol: HTTP
---
apiVersion: networking.istio.io/v1alpha3
kind: VirtualService
metadata:
  creationTimestamp: null
  name: http-istio-autogenerated-k8s-gateway
  namespace: apple
spec:
  gateways:
  - istio-system/gateway-istio-autogenerated-k8s-gateway
  hosts:
  - '*'
  http:
  - route:
    - destination:
        host: httpbin-apple.apple.svc.domain.suffix
        port:
          number: 80
---
apiVersion: networking.istio.io/v1alpha3
kind: VirtualService
metadata:
  creationTimestamp: null
  name: http-istio-autogenerated-k8s-gateway
  namespace: banana
spec:
  gateways:
  - istio-system/gateway-istio-autogenerated-k8s-gateway
  hosts:
  - '*'
  http:
  - route:
    - destination:
        host: httpbin-banana.banana.svc.domain.suffix
        port:
          number: 80
---
